File: play-msql-validation-anotation/app/jp/gmo/net/service/UserService.java
package jp.gmo.net.service;

import com.google.inject.ImplementedBy;

import jp.gmo.net.dto.RegisterFormData;
import jp.gmo.net.models.User;
import jp.gmo.net.models.UserDetail;
import jp.gmo.net.service.impl.UserServiceImpl;

@ImplementedBy(UserServiceImpl.class)
public interface UserService {
	public User getUserInfo(String email, String password);
	public void insertUser(RegisterFormData registerForm);
	public UserDetail getUserDetail(String email, String password);
}

File: hrmanager/app/service/UserService.scala
package service

import java.util.{ List => JList }

import com.google.inject.ImplementedBy

import entity.User
import javax.persistence.Entity
import javax.persistence.Table
import serviceImpl.UserServiceImpl
import forms._

@ImplementedBy(classOf[UserServiceImpl])
trait UserService {
  def findUserAll: JList[User]
  def findUserByEmail(email: String): User
  def saveByGoogle(user: UserGoogleForm): Int
  def updateUser(userUpdate: UpdateUserForm): Int
  def serviceLoginAccount(info : UserLoginAccountForm):Int
  
  def findUserSubtractEmail(email: String): JList[User]
   def registerUser(user: UpdateUserForm): Int
}